### ISSUE TITLE
Dynamic import broken for class and module attributes

### ISSUE THREAD

**Issue by has2k1** [machow/quartodoc#307]
With `dynamic=True` 

1. class attributes get the docstrings of their parent object
2. module attributes cannot be imported

This is a regression introduced in https://github.com/machow/quartodoc/commit/cd7fd5a9abcb60b15b8830590e94610c0afe6db6.

Here is how to reproduce it.

```python
from pathlib import Path
from quartodoc import get_object

text = '''
class AwesomeClass:
    """
    This is an awesome class
    """
    
    a: int
    b: dict
    """b is great"""

c: int = 1
d: dict = {}
"""d is great"""
'''

filepath = Path("tmp_test_file.py")
filepath.write_text(text)

obj_a = get_object("tmp_test_file:AwesomeClass.a", dynamic=True)
obj_b = get_object("tmp_test_file:AwesomeClass.b", dynamic=True)
# obj_c = get_object("tmp_test_file:c", dynamic=True)  # AttributeError
# obj_d = get_object("tmp_test_file:d", dynamic=True)  # AttributeError

print(obj_a.docstring and obj_a.docstring.value)
print(obj_b.docstring.value)
# print(obj_c.docstring and obj_a.docstring.value)
# print(obj_d.docstring.value)
```

```
This is an awesome class
This is an awesome class
```

The output should be the same when `dynamic=False`

```py
obj_a = get_object("tmp_test_file:AwesomeClass.a", dynamic=False)
obj_b = get_object("tmp_test_file:AwesomeClass.b", dynamic=False)
obj_c = get_object("tmp_test_file:c", dynamic=False)
obj_d = get_object("tmp_test_file:d", dynamic=False)

print(obj_a.docstring and obj_a.docstring.value)
print(obj_b.docstring.value)
print(obj_c.docstring and obj_a.docstring.value)
print(obj_d.docstring.value)
```

```
None
b is great
None
d is great
```

**Comment by machow** [comment]
Shoot, thanks for the example--I'm working on this now, and will add these to the test suite...

**Comment by machow** [comment]
From testing with quarto v0.6.6, things seem like this:

* regression: module attributes cannot be imported.
  - (note also encountered an error importing value-less class attributes, so this is a big regression).
* same behavior: class attributes get the docstrings of their parent object.
  - note that this only applies to class attributes with values (i.e. not purely an annotation)
  - the reason is that, when loading dynamically, quartodoc can't tell whether you meant to get the values docstring or not.

```python
class AwesomeClass:
    """
    This is an awesome class
    """
    
    # REGRESSION: couldn't dynamically import a or b ----
    a: int
    b: dict
    """b is great"""

# UNCHANGED BEHAVIOR: docstring for `1` or `{}` being used ----
# quartodoc can't tell whether you mean to use the dynamically imported
# objects docstring or not (e.g. if it were a function instead of `1`, people
# often want to use the function's docstring).
#
# For cases like d, we could also try to *statically* load the docstring ("d is great"),
# and if that works, then use that one?
c: int = 1
d: dict = {}
"""d is great"""
```

**Comment by machow** [comment]
I've added a fix and tests for the regression in https://github.com/machow/quartodoc/pull/309 -- do you mind checking to see if it works for you?!

**Comment by machow** [comment]
Should be resolved by #309 , definitely let me know if you run into issues -- happy to relook
